STACpopulator/input.py

```diff
@@ -58,21 +63,47 @@ def reset(self):
         """Reset the generator."""
         self.catalog_head = self.catalog
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator[Tuple[str, MutableMapping[str, Any]]]:
         """Return a generator walking a THREDDS data catalog for datasets."""
-        yield from self.catalog_head.datasets.items()
+        if self.catalog_head.datasets.items():
+            for item_name, ds in self.catalog_head.datasets.items():
+                attrs = self.extract_metadata(ds)
+                yield item_name, attrs
 
         if self._depth > 0:
             for name, ref in self.catalog_head.catalog_refs.items():
                 self.catalog_head = ref.follow()
                 self._depth -= 1
                 yield from self
 
+    def extract_metadata(self, ds: siphon.catalog.Dataset) -> MutableMapping[str, Any]:
+        # Get URL for NCML service
+        url = ds.access_urls["NCML"]
+
+        LOGGER.info("Requesting NcML dataset description")
+        r = requests.get(url)
```

Comment:
I've found that if you use this
```r = requests.get(url, params={"catalog": catalog, "dataset": dataset})```

You get more information in the response, in particular, you get a group called "THREDDSMetadata" with all the services offered by THREDDS.  The line below (access_urls) would not be needed.

Reply:
Try this: 

https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/ncml/birdhouse/testdata/xclim/cmip6/sic_SImon_CCCma-CanESM5_ssp245_r13i1p2f1_2020.nc?catalog=https%3A%2F%2Fpavics.ouranos.ca%2Ftwitcher%2Fows%2Fproxy%2Fthredds%2Fcatalog%2Fbirdhouse%2Ftestdata%2Fxclim%2Fcmip6%2Fcatalog.html&dataset=birdhouse%2Ftestdata%2Fxclim%2Fcmip6%2Fsic_SImon_CCCma-CanESM5_ssp245_r13i1p2f1_2020.nc

Reply:
It's the link provided on the THREDDS web page for the NcML service.